modules/Walking_module/src/WalkingModule.cpp

```diff
@@ -673,9 +417,56 @@ bool WalkingModule::updateModule()
 {
     std::lock_guard<std::mutex> guard(m_mutex);
 
-    if(m_robotState == WalkingFSM::Walking
-       || m_robotState == WalkingFSM::Stance
-       || m_robotState == WalkingFSM::OnTheFly)
+    if(m_robotState == WalkingFSM::Preparing)
+    {
+        bool motionDone = false;
+        if(!m_robotControlHelper->checkMotionDone(motionDone))
+        {
+            yError() << "[updateModule] Unable to check if the motion is done";
+            return false;
```

Comment:
Yes, you are right. In 705a65d instead of returning `false`, the controller goes in a `Stopped` state and the user has the possibility to prepare again.